synchronicity/synchronizer.py

```diff
@@ -376,10 +378,17 @@ def f_wrapped(*args, **kwargs):
 
                 return self._translate_out(res, interface)
 
+        if interface == Interface.ASYNC and inspect.iscoroutinefunction(f):
```

Comment:
This is arguably not very clean :D Preferably we'd make explicit "async def" function wrappers around all the async interface methods, but it was a bit of a pain to do that due to multiple layers of wrapping in the library at the moment

Reply:
That weirdness with `inspect.coroutinefunction` is why I am adding the `Awaitable` return annotation manually, to signal that the function returns a coroutine. The problem is not that annotations themselves aren't transferred, but that syncronicity's wrappers themselves are vanilla `def` functions returning coroutines, and as such can't be introspected to determine that they are "async".

Reply:
"Cleaned it up" now to use actual "async def" wrappers for async interfaces when the wrapped function is async. This will make `inspect.iscoroutinefunction` work (for async interfaces) as long as the underlying wrapped function is async (added a test for this as well).

Reply:
> as long as the underlying wrapped function is async (added a test for this as well).

yeah but i think it's a bit sketchy to rely on that? that's why i'd rather just take the position that synchronicity just transfers existing annotations but doesn't try to infer annotations in itself

Reply:
In my mind that's what I'm doing here, but I'd be happy to implement it differently if you have any ideas :)

Reply:
In my mind that's what I'm doing here, but I'd be happy to implement it differently if you have any ideas :)

Reply:
yeah you're right, it's really just setting the type here at this point